test/gem_notifications_tests.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'minitest/autorun'
+require_relative '../.github/workflows/scripts/slack_gem_notifications/notifications_methods'
+
+class GemNotifications < Minitest::Test
+  def test_check_for_updates
+    watched_gems = ["puma"]
+    assert check_for_updates(watched_gems)
+  end
+
+  def test_check_for_updates_if_empty
+    error = assert_raises(Exception) { check_for_updates([]) }
+    assert_equal("Nothing to see here! watched_gems cannot be empty", error.message)
+  end
+
+  def test_gem_list_empty
+    error = assert_raises(Exception) { gem_list_empty?([]) }
+    assert_equal("Nothing to see here! watched_gems cannot be empty", error.message)
+    assert_equal false, gem_list_empty?(['tyrannosaurus'])
+  end
+
+  def test_valid_gem_name
+    assert_equal true, is_valid_gem?("puma")
+    assert_equal false, is_valid_gem?("TrexRawr!")
```

Comment:
I see 2 problems here:

- The HTTParty gem appears to be used when these assertions are made, but `httparty` is not declared as a dev dependency for the project.
- We want to avoid making any real network calls to third party servers when the tests are ran.